nixos/modules/services/web-apps/cryptpad.nix

```diff
@@ -0,0 +1,213 @@
+      users.groups.cryptpad = {};
+
+      systemd.services.cryptpad = {
+        description = "Cryptpad service";
+        wantedBy = [ "multi-user.target" ];
+        after = [ "networking.target" ];
+        serviceConfig = {
+          User = "cryptpad";
+          Environment = [
+            "CRYPTPAD_CONFIG=${cryptpadConfigFile}"
+            "HOME=%S/cryptpad"
+          ];
+          ExecStart = "${cfg.package}/bin/cryptpad";
+          PrivateTmp = true;
+          Restart = "always";
+          StateDirectory = "cryptpad";
+          WorkingDirectory = "%S/cryptpad";
+        };
+      };
+    }
+    (mkIf cfg.confinement {
+      systemd.services.cryptpad = {
+        serviceConfig.BindReadOnlyPaths = [
+          cryptpadConfigFile
+          # apparently needs proc for workers management
+          "/proc"
+          "/dev/urandom"
+        ] ++ (if ! cfg.config.blockDailyCheck then [
+          "/etc/resolv.conf"
+          "/etc/hosts"
+        ] else []);
+        confinement = {
+          enable = true;
+          binSh = null;
+          mode = "chroot-only";
+        };
+      };
+    })
+    (mkIf cfg.configureNginx {
+      assertions = [
+        { assertion = cfg.config.httpUnsafeOrigin != "";
+          message = "services.cryptpad.config.httpUnsafeOrigin is required";
+        }
+        { assertion = strings.hasPrefix "https://" cfg.config.httpUnsafeOrigin;
+          message = "services.cryptpad.config.httpUnsafeOrigin must start with https://";
+        }
+        { assertion = isNull cfg.config.httpSafeOrigin || strings.hasPrefix "https://" cfg.config.httpSafeOrigin;
+          message = "services.cryptpad.config.httpSafeOrigin must start with https:// (or be unset)";
+        }
+      ];
+      services.nginx = {
+        enable = true;
+        recommendedTlsSettings = true;
+
+        # FIXME: Check / compare this with [Nginx module configuration in nixpkgs](https://github.com/NixOS/nixpkgs/blob/nixos-23.11/nixos/modules/services/web-servers/nginx/default.nix).
+        # Find out why Cryptpad has this in their documetation. Does this decrease the security of a Cryptpad install
+        # if not used?
+        # diffie-hellman parameters are used to negotiate keys for your session
+        # generate strong parameters using the following command
+        # ssl_dhparam /etc/nginx/dhparam.pem; # openssl dhparam -out /etc/nginx/dhparam.pem 4096
+        #
+        # sslDhparam = null;
+
+        # FIXME: Check / compare this with [Nginx module configuration in nixpkgs](https://github.com/NixOS/nixpkgs/blob/nixos-23.11/nixos/modules/services/web-servers/nginx/default.nix).
+        # Find out why Cryptpad has this in their documentation. Does this decrease the security of a Cryptpad install
+        # if not used?
+        # replace with the IP address of your resolver
+        # resolver 8.8.8.8 8.8.4.4 1.1.1.1 1.0.0.1 9.9.9.9 149.112.112.112 208.67.222.222 208.67.220.220;
+        #
+        # resolver = {};
+
```

Comment:
```suggestion
```
I've looked into this. Nginx allows you to specify built in resolver configuration, but I assume it falls back to the system resolver. Either way users can configure this through the [Nginx NixOS module settings](https://search.nixos.org/options?channel=23.11&show=services.nginx.resolver&from=0&size=50&sort=relevance&type=packages&query=nginx+resolver) so I think it's safe to remove this here.